fineract-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/domain/LoanProduct.java

```diff
@@ -711,6 +722,12 @@ public LoanProduct(final Fund fund, final String transactionProcessingStrategyCo
     }
 
     public void validateLoanProductPreSave() {
+        if (this.loanProductPaymentAllocationRules != null && loanProductPaymentAllocationRules.size() > 0
+                && !transactionProcessingStrategyCode.equals("advanced-repayment-allocation-strategy")) {
```

Comment:
Maybe "advanced-payment-allocation-strategy" would be better.

Reply:
fixed